### DISCUSSION TITLE
Is there anyway to generate cypress reports in slack

### DISCUSSION THREAD

**Discussion by bezba** [cypress-io/cypress#9240] [Questions and Help]
I wanna know if its possible to use slack webhook, to post a generated report by cypress. 
if I run cypress:run that generates a report at the end is there anyway of posting that into slack channel using a webhook?I have seen people doing it in Java, but havent seen any example using javascript.

**Comment by jennifer-shehane** [comment]
While this doesn't address the specifics of your question, I just wanted to mention that the Cypress Dashboard offers [Slack Integration](https://on.cypress.io/slack-integration), which will post Cypress test results to any slack channel you set up. https://on.cypress.io/slack-integration

If you have ideas for improving this feature, you can submit them in our Dashboard product board here: https://portal.productboard.com/cypress-io/1-cypress-dashboard

**Comment by bezba** [comment]
@jennifer-shehane  is it possible to have this run inside /support/index.js and then do a Post ?
`
```
Cypress.on("test:after:run", (test, runnable) => {
var xmlhttp = new XMLHttpRequest(),
        webhook_url = url-you-saved-from-before,
        myJSONStr= json-string-from-above;
    xmlhttp.open('POST', webhook_url, false);
    xmlhttp.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
    xmlhttp.send(MochaGenerated Report);
}
```

**Reply by bezba** [reply]
@jennifer-shehane I ended up doing it this way. When I run cypress run , it gives me a big summary report at the end. I still don't know how I can access that report. so i can get one big report .

```
after(()=>{
   let reportedTests
   let  url = SLACK_WEBHOOK
   const findTests = suite => {
    suite.tests.forEach(test => {
        if (test.state === "failed") {
            reportedTests = {
                "username": "Report notifier",
                "text": "New Report  <!here>", 
                "attachments": [ 
                  {
                    "color": "#FF0000",
                    "fields": [
                        {
                            "title" : test.parent.title + " "+ test.title,
                            "value" : test.state,
                            "short": true
                        }
                        
                    ]
                  
                    }
                ]
            }
              cy.request({
                method: 'POST',
                url: url,
                followRedirect: false,
                headers: {
                    'Content-Type': 'application/json'
                },
                body: reportedTests
            })
        }
  
```

**Comment by bahmutov** [comment]
The simplest way to get all test results so you can post them to Slack, or send anywhere really is to use our https://on.cypress.io/module-api. Here is a blog post that shows how to wrap Cypress NPM module API and post-process results https://glebbahmutov.com/blog/wrap-cypress-using-npm/

**Comment by bdimitrovski** [comment]
@bezba Perhaps a bit late to the party, but you can use this [script which I have developed](https://github.com/bdimitrovski/cypress-healthcheck) a while ago, to simply send test results back to Slack, without wrapping Cypress NPM module API.
